src/sidebar/components/search-status-bar.js

```diff
@@ -23,41 +27,90 @@ function SearchStatusBarController(store, rootThread) {
     );
   });
 
-  this.filterMatchCount = function() {
-    return visibleCount(thread());
+  const filterResults = () => {
+    const resultsCount = visibleCount(thread());
+    switch (resultsCount) {
+      case 0:
+        return 'No results for ' + filterQuery();
+      case 1:
+        return '1 search result';
+      default:
+        return resultsCount + ' search results';
+    }
   };
 
-  this.areAllAnnotationsVisible = function() {
-    if (store.getState().directLinkedGroupFetchFailed) {
+  const areAllAnnotationsVisible = () => {
+    if (useStore(store => store.getState().directLinkedGroupFetchFailed)) {
       return true;
     }
-    const selection = store.getState().selectedAnnotationMap;
+    const selection = useStore(store => store.getState().selectedAnnotationMap);
     if (!selection) {
       return false;
     }
     return Object.keys(selection).length > 0;
   };
 
-  this.filterQuery = function() {
-    return store.getState().filterQuery;
+  const filterQuery = () => {
+    return useStore(store => store.getState().filterQuery);
   };
 
-  this.filterActive = function() {
-    return !!store.getState().filterQuery;
+  const filterActive = () => {
+    return !!useStore(store => store.getState().filterQuery);
   };
 
-  this.onClearSelection = function() {
-    store.clearSelection();
-  };
+  const onClearSelection = useStore(store => store.clearSelection);
+
+  return (
+    <div className="search-status-bar">
+      {filterActive() && (
+        <div className="search-status-bar">
+          <button
+            className="primary-action-btn primary-action-btn--short"
+            onClick={onClearSelection}
+            title="Clear the search filter and show all annotations"
+          >
+            <i className="primary-action-btn__icon h-icon-close"></i>Clear
+            search
+          </button>
+          <span>{filterResults()}</span>
+        </div>
+      )}
+      {!filterActive() && areAllAnnotationsVisible() && (
+        <div className="search-status-bar">
+          <button
+            className="primary-action-btn primary-action-btn--short"
+            onClick={onClearSelection}
+            title="Clear the selection and show all annotations"
+          >
+            {(!selectedTab || selectedTab === uiConstants.TAB_ORPHANS) && (
+              <span>Show all annotations and notes</span>
+            )}
+            {selectedTab === uiConstants.TAB_ANNOTATIONS && (
+              <span>
```

Comment:
It doesn't look to me like the outer `span` is needed in the DOM as it doesn't have any semantic meaning or custom HTML attributes. If you need a wrapper element for a group of text nodes or elements purely so that the JSX is syntactically valid, you can use [`Fragment`](https://reactjs.org/docs/react-api.html#fragments) instead:

```js
<span><Fragment>item1 item2 item3</Fragment></span>
```

Would render as:

```html
<span>item1 item2 item3</span>
```

Search the existing client code for some example uses.